cmd/agent/api/internal/agent/agent.go

```diff
@@ -256,6 +258,72 @@ func streamLogs(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func streamEventPlatform(w http.ResponseWriter, r *http.Request) {
+	log.Info("Got a request for stream event platform.")
+	w.Header().Set("Transfer-Encoding", "chunked")
+
+	epMessageReceiver := epforwarder.GetGlobalReceiver()
+
+	flusher, ok := w.(http.Flusher)
+	if !ok {
+		log.Errorf("Expected a Flusher type, got: %v", w)
+		return
+	}
+
+	if epMessageReceiver == nil {
+		http.Error(w, "The agent is not running", 405)
+		flusher.Flush()
+		log.Info("Agent is not running - can't stream event platform")
+		return
+	}
+
+	if !epMessageReceiver.SetEnabled(true) {
+		http.Error(w, "Another client is already streaming event platform payloads.", 405)
+		flusher.Flush()
+		log.Info("Event platform payloads are already streaming. Dropping connection.")
+		return
+	}
+	defer epMessageReceiver.SetEnabled(false)
+
+	var filters diagnostic.Filters
+
+	if r.Body != http.NoBody {
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, log.Errorf("Error while reading HTTP request body: %s", err).Error(), 500)
+			return
+		}
+
+		if err := json.Unmarshal(body, &filters); err != nil {
+			http.Error(w, log.Errorf("Error while unmarshaling JSON from request body: %s", err).Error(), 500)
+			return
+		}
+	}
+
+	// Reset the `server_timeout` deadline for this connection as streaming holds the connection open.
+	conn := GetConnection(r)
+	_ = conn.SetDeadline(time.Time{})
+
+	done := make(chan struct{})
+	defer close(done)
+	logChan := epMessageReceiver.Filter(&filters, done)
+	flushTimer := time.NewTicker(time.Second)
+	for {
+		// Handlers for detecting a closed connection (from either the server or client)
+		select {
+		case <-w.(http.CloseNotifier).CloseNotify(): //nolint
+			return
+		case <-r.Context().Done():
+			return
+		case line := <-logChan:
+			fmt.Fprint(w, line)
```

Comment:
Is there any sensitive data here ? Password, API_KEY, configurations, ....
If so we should scrub the content.

Reply:
This is functionally identical to the `streamLogs` function right above it (see the other comment asking about a possible refactor). Since both use the same backend, any sensitive data scrubbing is happening before it is put into the `logChan`, in the `FormatMessage` functions that have been added